src/Lecture1.hs

```diff
@@ -67,7 +69,10 @@ sumOfSquares x y = error "TODO!"
 
 -}
 -- DON'T FORGET TO SPECIFY THE TYPE IN HERE
-lastDigit n = error "lastDigit: Not implemented!"
+lastDigit :: Integral a => a -> a
+lastDigit n
+    | n >= 0 && n<= 9 = n
+    | otherwise = lastDigit (mod (abs n) 10)
```

Comment:
You don't need a recursive call to `lastDigit` and a bounds check. `mod (abs n) 10` already returns the expected result 👍🏻